client/client_test.go

```diff
@@ -336,3 +338,25 @@ func TestResetSession(t *testing.T) {
 func TestAdditionalHeaders(t *testing.T) {
 	testAdditionalHeaders(t, clientFactory)
 }
+
+func TestAsyncQuery(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case ServiceAccountLoginURLSuffix:
+			_, _ = w.Write(utils.GetAuthResponse(10000))
+		default:
+			if r.URL.Query().Get("async") != "true" {
+				t.Errorf("Did not set async query parameter to true in async context")
+			}
+			w.WriteHeader(http.StatusOK)
+		}
```

Comment:
Let's also flip some variable once we're here. Otherwise if we never send an async query this test would still pass.